homeassistant/components/p1_monitor/const.py

```diff
@@ -0,0 +1,229 @@
+"""Constants for the P1 Monitor integration."""
+from __future__ import annotations
+
+import logging
+from typing import Final
+
+from homeassistant.components.sensor import (
+    STATE_CLASS_MEASUREMENT,
+    STATE_CLASS_TOTAL_INCREASING,
+    SensorEntityDescription,
+)
+from homeassistant.const import (
+    CURRENCY_EURO,
+    DEVICE_CLASS_CURRENT,
+    DEVICE_CLASS_ENERGY,
+    DEVICE_CLASS_GAS,
+    DEVICE_CLASS_POWER,
+    DEVICE_CLASS_VOLTAGE,
+    ELECTRIC_CURRENT_AMPERE,
+    ELECTRIC_POTENTIAL_VOLT,
+    ENERGY_KILO_WATT_HOUR,
+    POWER_WATT,
+    VOLUME_CUBIC_METERS,
+)
+from homeassistant.util import dt
+
+DOMAIN: Final = "p1_monitor"
+LOGGER = logging.getLogger(__package__)
+DEFAULT_TIME_BETWEEN_UPDATE = 5
+
+ATTR_ENTRY_TYPE: Final = "entry_type"
+ENTRY_TYPE_SERVICE: Final = "service"
+
+CONF_TIME_BETWEEN_UPDATE = "time_between_update"
+
+SERVICE_SMARTMETER: Final = "smartmeter"
+SERVICE_PHASES: Final = "phases"
+SERVICE_SETTINGS: Final = "settings"
+
+SERVICES: dict[str, str] = {
+    SERVICE_SMARTMETER: "SmartMeter",
+    SERVICE_PHASES: "Phases",
+    SERVICE_SETTINGS: "Settings",
+}
+
+SENSORS: dict[str, list[SensorEntityDescription]] = {
+    SERVICE_SMARTMETER: [
+        SensorEntityDescription(
+            key="gas_consumption",
+            name="Gas Consumption",
+            icon="mdi:fire",
```

Comment:
icon should not be set if you set a device class

Reply:
There was an issue in the frontend loading the icon for this particular device_class. https://github.com/home-assistant/frontend/pull/9830